CMakeLists.txt

```
@@ -37,6 +37,7 @@ option(NEOFOAM_BUILD_TESTS "Build the unit tests" OFF)
 option(NEOFOAM_BUILD_BENCHMARKS "Build benchmarks" OFF)
 option(NEOFOAM_BUILD_DOC "Build documentation" OFF)
 option(NEOFOAM_WITH_SUNDIALS "Build NeoFOAM with Sundials support [currently required]" ON)
+option(NEOFOAM_WITH_PETSC "Build NeoFOAM with petsc support" ON)
```

Let's keep it as an optional library for now. 

```suggestion
option(NEOFOAM_WITH_PETSC "Build NeoFOAM with petsc support" OFF)
```